server/src/main/java/org/opensearch/action/termvectors/TransportMultiTermVectorsAction.java

```diff
@@ -123,6 +126,9 @@ protected void doExecute(Task task, final MultiTermVectorsRequest request, final
                 .shardId(clusterState, concreteSingleIndex, termVectorsRequest.id(), termVectorsRequest.routing());
             MultiTermVectorsShardRequest shardRequest = shardRequests.get(shardId);
             if (shardRequest == null) {
+                if (shouldForcePrimaryRouting(clusterState, request.realtime, request.preference, concreteSingleIndex)) {
```

Comment:
Rebase this over #9536 once it's merged. Changing parameter to metadata instead of clusterState.

Reply:
Done